coordinator/src/collaborative_revert.rs

```diff
@@ -25,257 +26,280 @@ use dlc_manager::subchannel::LnDlcChannelSigner;
+        None => {
+            let fund_txo_sat = Decimal::from(fund_txo_sat);
 
-    let settlement_amount = position
-        .calculate_settlement_amount(revert_params.price)
-        .context("Could not calculate settlement amount")?;
-
-    let trade_settled = sub_channel.fund_value_satoshis == channel_details.channel_value_satoshis;
-
-    // There is no easy way to get the total tx fee for all subchannel transactions, hence, we
-    // estimate it. This transaction fee is shared among both users fairly
-    let dlc_channel_fee = calculate_dlc_channel_tx_fees(
-        trade_settled,
-        sub_channel.fund_value_satoshis,
-        channel_details.inbound_capacity_msat / 1000,
-        channel_details.outbound_capacity_msat / 1000,
-        position.trader_margin as u64,
-        position.coordinator_margin as u64,
-        channel_details.unspendable_punishment_reserve.unwrap_or(0),
-    )?;
-
-    // Coordinator's amount is the total channel's value (fund_value_satoshis) whatever the taker
-    // had (inbound_capacity), the taker's PnL (settlement_amount) and the transaction fee
-    let coordinator_amount = match trade_settled {
-        false => {
-            sub_channel.fund_value_satoshis as i64
-                - (channel_details.inbound_capacity_msat / 1000) as i64
-                - settlement_amount as i64
-                - (dlc_channel_fee as f64 / 2.0) as i64
-                - channel_details.unspendable_punishment_reserve.unwrap_or(0) as i64
+            let reserve = fund_txo_sat / Decimal::ONE_HUNDRED;
+
+            // The minimum reserve is _at least_ 1_000 sats.
+            let reserve = reserve.max(Decimal::ONE_THOUSAND);
+
+            reserve.to_u64().expect("Reserve to fit into u64")
         }
-        true => {
-            sub_channel.fund_value_satoshis as i64
-                - (channel_details.inbound_capacity_msat / 1000) as i64
-                - channel_details.unspendable_punishment_reserve.unwrap_or(0) as i64
+    };
+
+    let is_channel_split = match channel_details_option {
+        Some(ChannelDetails {
+            funding_txo: Some(funding_txo),
+            original_funding_outpoint: Some(original_funding_txo),
+            ..
+        }) => {
+            // The channel _is_ split if the `original_funding_txo` and the `funding_txo` differ.
+            original_funding_txo != funding_txo
+        }
+        Some(ChannelDetails {
+            funding_txo: Some(_),
+            original_funding_outpoint: None,
+            ..
+        }) => {
+            // The channel is _not_ split if the `original_funding_txo` has not been set.
+            false
+        }
+        Some(ChannelDetails {
+            funding_txo: None, ..
+        }) => {
+            bail!("Cannot collaboratively revert channel without funding TXO");
+        }
+        None => {
+            let params = without_channel_details_params.context(
```

Comment:
nit: I guess if we do not have a split channel, we could simple assume that the inbound and outbound capacity have the correct values and we could return always true here.

The only reason I added this "flag" was that in some scenarios the inbound/outbound liquidity already included the position funds and sometimes not. The easiest way to check that was by comparing the ln balance and the sub channel capacity. If they match the amount was already settled into the lightning channel.

Reply:
> nit: I guess if we do not have a split channel, we could simple assume that the inbound and outbound capacity have the correct values and we could return always true here.

I think you mean `false` provided that I action [this](https://github.com/get10101/10101/pull/1607#discussion_r1398853853) comment of yours. And I tend to agree. My first version was defaulting to `false` here. I am happy to revert to that.

With all your comments I think we can go back to a version without optional parameters, which is nice.